src/app/zones/views/ZonesList/ZonesListTable/EditZone/EditZone.tsx

```diff
@@ -1,76 +1,76 @@
-import { Row, Col, Textarea } from "@canonical/react-components";
+import type { ReactElement } from "react";
+
+import { Notification, Spinner, Textarea } from "@canonical/react-components";
 import { useQueryClient } from "@tanstack/react-query";
 
 import { useGetZone, useUpdateZone } from "@/app/api/query/zones";
-import type { UpdateZoneError } from "@/app/apiclient";
+import type { UpdateZoneError, ZoneRequest } from "@/app/apiclient";
 import { getZoneQueryKey } from "@/app/apiclient/@tanstack/react-query.gen";
 import FormikField from "@/app/base/components/FormikField";
 import FormikForm from "@/app/base/components/FormikForm";
 
-type Props = {
+type EditZoneProps = {
   id: number;
   closeForm: () => void;
 };
 
-export type CreateZoneValues = {
-  description: string;
-  name: string;
-};
-
-const EditZone = ({ id, closeForm }: Props): React.ReactElement | null => {
+const EditZone = ({ id, closeForm }: EditZoneProps): ReactElement => {
   const queryClient = useQueryClient();
-  const { data: zone } = useGetZone({ path: { zone_id: id } });
+  const zone = useGetZone({ path: { zone_id: id } });
 
   const editZone = useUpdateZone();
 
-  if (zone) {
-    return (
-      <FormikForm<CreateZoneValues, UpdateZoneError>
-        aria-label="Edit AZ"
-        errors={editZone.error}
-        initialValues={{
-          description: zone.description,
-          name: zone.name,
-        }}
-        onCancel={closeForm}
-        onSubmit={(values) => {
-          editZone.mutate({
-            body: { name: values.name, description: values.description },
-            path: { zone_id: id },
-          });
-        }}
-        onSuccess={() => {
-          queryClient
-            .invalidateQueries({
-              queryKey: getZoneQueryKey({
-                path: { zone_id: id },
-              }),
-            })
-            .then(closeForm);
-        }}
-        saved={editZone.isSuccess}
-        saving={editZone.isPending}
-        submitLabel="Update AZ"
-      >
-        <Row>
-          <Col size={12}>
-            <FormikField
-              label="Name"
-              name="name"
-              placeholder="Name"
-              type="text"
-            />
-            <FormikField
-              component={Textarea}
-              label="Description"
-              name="description"
-            />
-          </Col>
-        </Row>
-      </FormikForm>
-    );
-  }
-  return null;
+  return (
+    <>
+      {zone.isPending && <Spinner text="Loading..." />}
+      {zone.isError && (
+        <Notification data-testid="no-such-pool-error" severity="negative">
+          AZ with id {id} does not exist.
+        </Notification>
+      )}
+      {zone.isSuccess && zone.data && (
```

Comment:
Nice :+1: just make sure there are test cases for these so our branch coverage stays up